test/jdk/java/nio/channels/DatagramChannel/AdaptorMulticasting.java

```diff
@@ -421,8 +427,22 @@ static void testSendReceive(MulticastSocket s, InetAddress group) throws IOExcep
 
         // receive message
         s.setSoTimeout(0);
-        p = new DatagramPacket(new byte[1024], 100);
-        s.receive(p);
+        do {
+            p = new DatagramPacket(new byte[1024], 100);
+            s.receive(p);
+            if (getPort(p.getSocketAddress()) == getPort(s.getLocalSocketAddress())) {
```

Comment:
I assume this can be simplified to `if (p.getPort() == s.getLocalPort())`, meaning no need to use SocketAddress here.